src/utils/datetime/validator.test.ts

```diff
@@ -99,48 +99,57 @@ describe('the datetime validator', () => {
     ).toBeFalsy()
   })
   it('should be strict on date formats', function() {
-    expect(isValid('1999-02-09 23:00:0', 'YYYY-MM-DD HH:mm:ss')).toBeFalsy()
+    const strict = true
     expect(
-      isValid('1999-02-09 12:00:00 PM -1', 'YYYY-MM-DD hh:mm:ss a ZZ')
+      isValid('1999-02-09 23:00:0', 'YYYY-MM-DD HH:mm:ss', strict)
     ).toBeFalsy()
     expect(
-      isValid('09/02/1999 23:00:00.0', 'DD/MM/YYYY HH:mm:ss.sss')
+      isValid('1999-02-09 12:00:00 PM -1', 'YYYY-MM-DD hh:mm:ss a ZZ', strict)
     ).toBeFalsy()
     expect(
-      isValid('09/02/1999 11:00:00.000 A', 'DD/MM/YYYY hh:mm:ss.sss a')
+      isValid('09/02/1999 23:00:00.0', 'DD/MM/YYYY HH:mm:ss.sss', strict)
     ).toBeFalsy()
     expect(
-      isValid('02/09/1999 23:00:00.00', 'MM/DD/YYYY HH:mm:ss.sss')
+      isValid('09/02/1999 11:00:00.000 A', 'DD/MM/YYYY hh:mm:ss.sss a', strict)
     ).toBeFalsy()
     expect(
-      isValid('02/09/1999 11:00:00.000 A', 'MM/DD/YYYY hh:mm:ss.sss a')
+      isValid('02/09/1999 23:00:00.00', 'MM/DD/YYYY HH:mm:ss.sss', strict)
     ).toBeFalsy()
-    expect(isValid('1999/02/09 23:00:0', 'YYYY/MM/DD HH:mm:ss')).toBeFalsy()
-    expect(isValid('1999/02/09 11:00', 'YYYY/MM/DD hh:mm:ss a')).toBeFalsy()
-    expect(isValid('23:0', 'HH:mm')).toBeFalsy()
-    expect(isValid('11:00 A', 'hh:mm a')).toBeFalsy()
-    expect(isValid('23:00:0', 'HH:mm:ss')).toBeFalsy()
-    expect(isValid('11:00:0 A', 'hh:mm:ss a')).toBeFalsy()
-    expect(isValid('23:00:00 -1', 'HH:mm:ss ZZ')).toBeFalsy()
-    expect(isValid('11:00:00 A -1', 'hh:mm:ss a ZZ')).toBeFalsy()
-    expect(isValid('23:00:00.0', 'HH:mm:ss.sss')).toBeFalsy()
-    expect(isValid('11:00:00.0 A', 'hh:mm:ss.sss a')).toBeFalsy()
     expect(
-      isValid('February 9, 1999 23:00:0', 'MMMM D, YYYY HH:mm:ss')
+      isValid('02/09/1999 11:00:00.000 A', 'MM/DD/YYYY hh:mm:ss.sss a', strict)
     ).toBeFalsy()
     expect(
-      isValid('February 9, 1999 11:00:0 PM', 'MMMM D, YYYY hh:mm:ss a')
+      isValid('1999/02/09 23:00:0', 'YYYY/MM/DD HH:mm:ss', strict)
+    ).toBeFalsy()
+    expect(
+      isValid('1999/02/09 11:00', 'YYYY/MM/DD hh:mm:ss a', strict)
+    ).toBeFalsy()
+    expect(isValid('23:0', 'HH:mm', strict)).toBeFalsy()
+    expect(isValid('11:00 A', 'hh:mm a', strict)).toBeFalsy()
+    expect(isValid('23:00:0', 'HH:mm:ss', strict)).toBeFalsy()
+    expect(isValid('11:00:0 A', 'hh:mm:ss a', strict)).toBeFalsy()
+    expect(isValid('23:00:00 -1', 'HH:mm:ss ZZ', strict)).toBeFalsy()
+    expect(isValid('11:00:00 A -1', 'hh:mm:ss a ZZ', strict)).toBeFalsy()
+    expect(isValid('23:00:00.0', 'HH:mm:ss.sss', strict)).toBeFalsy()
+    expect(isValid('11:00:00.0 A', 'hh:mm:ss.sss a', strict)).toBeFalsy()
+    expect(
+      isValid('February 9, 1999 23:00:0', 'MMMM D, YYYY HH:mm:ss', strict)
+    ).toBeFalsy()
+    expect(
+      isValid('February 9, 1999 11:00:0 PM', 'MMMM D, YYYY hh:mm:ss a', strict)
```

Comment:
All this diff is just because prettier and git. All I am adding is the `strict` parameter.